### PR TITLE
Make label width configurable

### PR DESCRIPTION
**Pull request by jasongrout**
This implements the ideas behind #1373. The idea is that the current label width is still too brittle - we have a min and a max width, but it's too hard to get something much smaller or larger. This introduces a description_width style attribute which lets you easily set the width. As part of this, I eliminated the max-width on the label, since now it is very easy to extend the width of a label. For example, `style=dict(description_width='initial')` makes the label be exactly as long as the text.

This also affects the checkbox indent. @maxnoy - what do you think about `description_width` affecting the checkbox `indent` width? It seems a *bit* confusing that `description_width` affects the width of the indent, not the text.

```python
from ipywidgets import *
style=dict(description_width='50px')
widgets = []
widgets.append(IntSlider(description='slide', style=style))
widgets.append(Dropdown(description='drop', style=style))
widgets.append(Checkbox(description='check', style=style))
VBox(widgets)
```

screen shot 2017-05-22 at 11 48 04 am

### CODE REVIEW

File: ipywidgets/widgets/domwidget.py [hunk 1]
```diff
@@ -37,8 +38,15 @@ def remove_class(self, className):
         return self
 
 
+class DescriptionStyle(Style, Widget):
+    """Button style widget."""
+    _model_name = Unicode('SliderStyleModel').tag(sync=True)
```
Comment (blink1073) [inline ipywidgets/widgets/domwidget.py]: misnamed?

Comment (jasongrout) [inline ipywidgets/widgets/domwidget.py]: Yep, thanks.

**Comment by jasongrout** [comment]
Opening another issue for discussion on the label width: #1377.

**Comment by maxnoy** [comment]
@jasongrout, this seems like very useful functionality. Maybe if there was a notion of description_width vs content_width, it would be a bit clearer?

**Comment by jasongrout** [comment]
I think we usually have the widgets automatically expand the content width, so there is a correlation between widget width and content width (i.e., to make the content wider, just make the widget wider).

Can we just handle this in documentation, with the explanation that the checkbox description is on the right?

**Pull request merged**